src/catalog/database_catalog.cpp

```diff
@@ -203,11 +209,72 @@ void DatabaseCatalog::Bootstrap(const common::ManagedPointer<transaction::Transa
+                            postgres::FK_SRC_TABLE_OID_INDEX_OID, "fk_constraint_src_table_oid_index",
+                            postgres::Builder::GetFKConstraintSrcTableOidIndexSchema(db_oid_));
+  TERRIER_ASSERT(retval, "Bootstrap operations should not fail");
+  retval = SetIndexPointer(txn, postgres::FK_SRC_TABLE_OID_INDEX_OID, fk_constraints_src_table_oid_index_);
+  TERRIER_ASSERT(retval, "Bootstrap operations should not fail");
+
+  retval = CreateIndexEntry(txn, postgres::NAMESPACE_CATALOG_NAMESPACE_OID, postgres::FK_TABLE_OID,
+                            postgres::FK_REF_TABLE_OID_INDEX_OID, "fk_constraint_ref_table_oid_index",
+                            postgres::Builder::GetFKConstraintRefTableOidIndexSchema(db_oid_));
+  TERRIER_ASSERT(retval, "Bootstrap operations should not fail");
+  retval = SetIndexPointer(txn, postgres::FK_REF_TABLE_OID_INDEX_OID, fk_constraints_ref_table_oid_index_);
+  TERRIER_ASSERT(retval, "Bootstrap operations should not fail");
+
+  ////
+  retval = CreateTableEntry(txn, postgres::CHECK_TABLE_OID, postgres::NAMESPACE_CATALOG_NAMESPACE_OID,
+                            "check_constraint", postgres::Builder::GetCheckConstraintTableSchema());
+  TERRIER_ASSERT(retval, "Bootstrap operations should not fail");
+  retval = SetTablePointer(txn, postgres::CHECK_TABLE_OID, check_constraints_);
   TERRIER_ASSERT(retval, "Bootstrap operations should not fail");
-  retval = SetIndexPointer(txn, postgres::CONSTRAINT_FOREIGNTABLE_INDEX_OID, constraints_foreigntable_index_);
+
+  retval = CreateIndexEntry(txn, postgres::NAMESPACE_CATALOG_NAMESPACE_OID, postgres::CHECK_TABLE_OID,
+                            postgres::CHECK_OID_INDEX_OID, "check_constraint_oid_index",
+                            postgres::Builder::GetCheckConstraintOidIndexSchema(db_oid_));
+  TERRIER_ASSERT(retval, "Bootstrap operations should not fail");
+  retval = SetIndexPointer(txn, postgres::CHECK_OID_INDEX_OID, check_constraints_oid_index_);
+  TERRIER_ASSERT(retval, "Bootstrap operations should not fail");
+  ////
+  retval = CreateTableEntry(txn, postgres::EXCLUSION_TABLE_OID, postgres::NAMESPACE_CATALOG_NAMESPACE_OID,
+                            "exclusion_constraint", postgres::Builder::GetExclusionConstraintTableSchema());
+  TERRIER_ASSERT(retval, "Bootstrap operations should not fail");
+  retval = SetTablePointer(txn, postgres::EXCLUSION_TABLE_OID, exclusion_constraints_);
+  TERRIER_ASSERT(retval, "Bootstrap operations should not fail");
+
+  retval = CreateIndexEntry(txn, postgres::NAMESPACE_CATALOG_NAMESPACE_OID, postgres::EXCLUSION_TABLE_OID,
+                            postgres::EXCLUSION_OID_INDEX_OID, "exclusion_constraint_oid_index",
+                            postgres::Builder::GetExclusionConstraintOidIndexSchema(db_oid_));
+  TERRIER_ASSERT(retval, "Bootstrap operations should not fail");
+  retval = SetIndexPointer(txn, postgres::EXCLUSION_OID_INDEX_OID, exclusion_constraints_oid_index_);
```

Comment:
As your OIDs start from 91, consider moving the bootstrap code after `pg_proc`. Currently the entries in `pg_class` are a little out of order.